php/ng/modules.sls

```diff
@@ -0,0 +1,23 @@
+{% from "php/ng/map.jinja" import php with context %}
+
+{% set modules = salt['pillar.get']('php:ng:modules') or [] %}
+{% set base_name = 'php.ng.' %}
+{% set existing_states = salt['cp.list_states']() %}
+
+{% set includes = [] %}
+{% set install = [] %}
+
+{% for module in modules %}
+{%   set state = base_name + module %}
```

Comment:
While `+` works, `~` is the [string concatenation operator in Jinja](http://jinja.pocoo.org/docs/2.10/templates/#other-operators).

Reply:
I didn't know about that... but not exactly "string concatenation" operator, but it converts its operands to strings first.

As module is something from user (pillars), it might not be a string, so yeah, changed to `~`.

If we were sure module were already a string, I would have stayed with `+`, like all there is in the other files.

Reply:
@ze42 Fair enough, not "string concatenation" specifically!  But from the [same page](http://jinja.pocoo.org/docs/2.10/templates/#math) regarding `+`:

> `+`
> Adds two objects together. Usually the objects are numbers, but if both are strings or lists, you _can_ concatenate them this way. This, however, **is not the preferred way to concatenate strings!** For string concatenation, have a look-see at the `~` operator.

Emphasis above is mine.

Reply:
If you want to go that way, you can make a PR, converting all 538 other instances of the `+` usage, by the `~` that is used twice so far (t. (that one being the third instance of `~` for string concat in that repo)

Difference between **prefered** (by those writing jinja), and **widely used** by those writing salt formula.

I tend to prefer `+`, as it looks more like what you'd do in python.

Anyway, there already is a commit having a `~` here.

Reply:
@ze42 Thanks for your feedback.  There's a wider picture at play.  Over the last few months, there have been a large number of discussions about SaltStack Formulas as a whole.  There are numerous complaints about the quality of the formulas, from many different angles.  I'm not going to repeat anything here but a general indication of changes being considered are outlined [in this issue](https://github.com/saltstack-formulas/template-formula/issues/21).  Realistically, there aren't enough hands on deck to push through all of these changes, across all of the formulas.  However, the structure is coming together so that some of the large steps can be taken org-wide.

> If you want to go that way, you can make a PR, converting all 538 other instances of the `+` usage, by the `~` that is used twice so far (t. (that one being the third instance of `~` for string concat in that repo)
> 

This really isn't difficult but it isn't fair on formula consumers at the moment.  If this formula ever gets `semantic-release`, this can be part of the first major overhaul.

> Difference between **prefered** (by those writing jinja), and **widely used** by those writing salt formula.
> 
> I tend to prefer `+`, as it looks more like what you'd do in python.
> 

For Jinja, we use Jinja best practices.  `~` is being used consistently across many of the major formulas in this org.

> Anyway, there already is a commit having a `~` here.

Thank you for making that change.